### PR TITLE
ResponseHeadersMiddleware: Add trace id header

### PR DESCRIPTION
**Pull request by daiiniel**
Add traceId header to responses if the Current Activity is not null to track requests in logs and trace explorers.

### CODE REVIEW

File: tests/Hosting.Services.Web.UnitTests/Middlewares/ResponseHeadersMiddlewareTests.cs [hunk 1]
```diff
@@ -15,6 +16,10 @@ namespace Hosting.Services.Web.UnitTests
 	[TestClass]
 	public class ResponseHeadersMiddlewareTests
 	{
+		private const string MachineHeaderName = "X-Machine";
+		private const string BuildVersionHeaderName = "X-Machine";
+		private const string TraceIdHeaderName = "X-Machine";
```
Comment (AndreyTretyak) [inline tests/Hosting.Services.Web.UnitTests/Middlewares/ResponseHeadersMiddlewareTests.cs]: It looks like thos two consts need to have diffirent values, or maybe get it from the cluss you are testing, by making them internal there.

Comment (daiiniel) [reply tests/Hosting.Services.Web.UnitTests/Middlewares/ResponseHeadersMiddlewareTests.cs]: Yeah thanks, I did not push latest changes :/

Comment (daiiniel) [reply tests/Hosting.Services.Web.UnitTests/Middlewares/ResponseHeadersMiddlewareTests.cs]: Fixed now

File: src/Hosting.Services.Web/Middlewares/ResponseHeadersMiddleware.cs [hunk 2]
```diff
@@ -25,6 +26,12 @@ private Task SetResponseHeaders(object state)
 			HttpResponse response = (HttpResponse)state;
 			response.Headers.Add("X-Machine", m_context.MachineId);
 			response.Headers.Add("X-BuildVersion", m_context.BuildVersion); //Renamed from X-OfficeVersion
+
+			if (Activity.Current != null)
+			{
+				response.Headers.Add("X-TraceId", Activity.Current.TraceId.ToString());
```
Comment (artempushkin) [inline src/Hosting.Services.Web/Middlewares/ResponseHeadersMiddleware.cs:32]: Isn't it available as a standard header?

Comment (daiiniel) [reply src/Hosting.Services.Web/Middlewares/ResponseHeadersMiddleware.cs:32]: AT least not for our service

image

Comment (artempushkin) [reply src/Hosting.Services.Web/Middlewares/ResponseHeadersMiddleware.cs:32]: I've found this from a quick search: https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/trace/getting-started-aspnetcore/README.md
Admittedly, the instrumentation is still pre-release, and looks like there are some performance concerns, but I'd rather go along with standardization effort than create a custom solution.

**Pull request closed without merging**